Data/Attoparsec/ByteString.hs

```diff
@@ -226,7 +226,8 @@ maybeResult _            = Nothing
 -- | Convert a 'Result' value to an 'Either' value. A 'T.Partial'
 -- result is treated as failure.
 eitherResult :: Result r -> Either String r
-eitherResult (T.Done _ r)        = Right r
-eitherResult (T.Fail _ [] msg)   = Left msg
-eitherResult (T.Fail _ ctxs msg) = Left (intercalate " > " ctxs ++ ": " ++ msg)
-eitherResult _                   = Left "Result: incomplete input"
+eitherResult result = case feed result B.empty of
+    T.Done _ r        -> Right r
+    T.Fail _ [] msg   -> Left msg
+    T.Fail _ ctxs msg -> Left (intercalate " > " ctxs ++ ": " ++ msg)
+    T.Partial _       -> error "feed result mempty = Partial _"
```

Comment:
This seems like a less understandable error than the original. Perhaps something like "eitherResult: Partial" would be less ambiguous?

Reply:
I don't mind changing it, but just to make sure we're on the same page - this is not a substitute for the error message that was previously returned when the parse result was partial. This is an error message that will never be seen.